Add device_paths

deepakala-k opened the pull request:
Remove the changes from the DTS file to insert the device paths It needs to be split into two commits.

SwethaParasa commented on data/p10/target_types_obmc.xml:
```diff
@@ -194,7 +204,7 @@
     <targetType>
         <id>chip-ocmb</id>
-        <parent>chip</parent>
+        <parent>scommable-chip</parent>
```
Location code is defined for chip target, but parent of ocmb-chip is changed to scommable-chip, will this miss Location code attribute for ocmb?

deepakala-k replied:
the new scommable-chip is child of chip. So it will have all the attributes of the chip and have the device paths in addition to it

dhruvibm replied:
Please use the same target types as hostboot, we will find ways to address usecaese, and creating additional layers for few attributes will create unwanted dependancies

deepakala-k replied:
done. removed the extra target.

devenrao commented on data/p10/attribute_types_obmc.xml:
```diff
@@ -661,64 +661,77 @@
-            pu  = generic processor
-        </description>
-        <id>FAPI_NAME</id>
+        <description>Sbefifo copath for scom operations</description>
```
typo copath?

devenrao commented on data/p10/target_types_obmc.xml:
```diff
@@ -52,33 +52,43 @@
         <parent>chip-processor</parent>
     </targetType>
     <targetType>
+        <id>scommable-chip</id>
```
is the id value correct scommable-chip?

deepakala-k replied:
Yes, I created the new id

deepakala-k committed:
Add hw access device paths at the MRW processing stage

For scom and cfam access, we need the backend device paths for the specific
chip. Adding the device paths based on a map corresponding to respective proc
or ocmb chips.

Tested and the generated xml contains the device paths as per the chip.

Signed-off-by: Deepa Karthikeyan <deepakala.karthikeyan@ibm.com>

deepakala-k committed:
Use device paths populated at the MRW processed stage

Update the filter to include the device paths, so that it gets added into the device tree.

Tested and works as expected.

Signed-off-by: Deepa Karthikeyan <deepakala.karthikeyan@ibm.com>

dhruvibm commented on data/p10/attribute_types_obmc.xml:
```diff
@@ -879,4 +892,37 @@
+        <hasStringConversion></hasStringConversion>
+        <id>SELECTED_BACKEND</id>
+        <no_export></no_export>
+        <persistency>non-volatile</persistency>
```
This should be a volatile attribute

deepakala-k replied:
done

dhruvibm commented on data/p10/attribute_types_obmc.xml:
```diff
@@ -879,4 +892,37 @@
+    <attribute>
+        <description>Type of selected backend</description>
+        <hasStringConversion></hasStringConversion>
+        <id>SELECTED_BACKEND</id>
```
Backend is valid when we had multiple device trees, I think HW_ACCESS_METHOD may be a better term

dhruvibm commented on data/p10/attribute_types_obmc.xml:
```diff
@@ -879,4 +892,37 @@
+            <value>0</value>
+        </enumerator>
+        <enumerator>
+            <name>KERNEL</name>
```
DIRECT or DIRECT_ACCESS instead of KERNEL

dhruvibm commented on data/p10/target_types_obmc.xml:
```diff
@@ -52,33 +52,43 @@
         <parent>chip-processor</parent>
     </targetType>
     <targetType>
+        <id>scommable-chip</id>
```
chip should be parent of chip-processor

deepakala-k replied:
chip
|
--> scommable-chip
       |
       --> chip-processor

So chip-processor has everything along with chip's attributes and it is scommable. tpm does not have the sbefifo and direct access path. So I introduced a new level in-betweem

deepakala-k replied:
removed the extra target

dhruvibm commented on scripts/processMrw.pl:
```diff
@@ -225,9 +250,8 @@ sub printUsage
-        -build <hb | fsp | bmc> = hb  - process HB targets only (the default)
-                                  fsp - process FSP targets in addition to HB targets
-                                  bmc - process BMC targets in addition to HB targets
+        -build <hb | bmc> = hb  - process HB targets only (the default)
```
let FSP be there and bmc is an additional argument

deepakala-k replied:
done
